src/Apps/Order/Routes/Shipping2/ShippingContext.ts

```diff
@@ -1,20 +1,342 @@
+    fulfillmentDetails: FulfillmentValues["attributes"] | null
+    fulfillmentType: FulfillmentValues["fulfillmentType"] | null
+    isArtsyShipping?: boolean
+    shippingQuotes?: Array<{ id: string; isSelected: boolean }>
+  }
+  initialValues: {
+    fulfillmentDetails: FulfillmentValues
+    shippingQuotes: {
+      selectedShippingQuoteId: any
+    }
+  }
+  step: ShippingRouteStep
+  helpers: {
+    fulfillmentDetails: FulfillmentHelpers & {
+      setFulfillmentFormHelpers: (helpers: FulfillmentHelpers) => void
+    }
+  }
 }
 
 export const ShippingContext = createContext<ShippingContextProps>({} as any)
 
 export const useShippingContext = () => {
   return useContext(ShippingContext)
 }
+
+// Get information from the order and user, savedOrderData (shared, computed context about order)
+// initial values for forms, and the current step in the shipping route
+// values for forms
+export const useLoadOrder = (props: ShippingProps): ShippingContextProps => {
+  const savedOrderData = useLoadComputedData(props)
+  const initialValues = getInitialValues(props, savedOrderData)
+
+  const {
+    fulfillmentType: savedFulfillmentType,
+
+    isArtsyShipping,
+  } = savedOrderData
+  const selectedShippingQuoteId =
+    initialValues.shippingQuotes.selectedShippingQuoteId
+
+  const targetStep: ShippingRouteStep = useMemo(() => {
+    if (!savedFulfillmentType) {
+      return "fulfillment_details"
+    }
+    if (isArtsyShipping && !selectedShippingQuoteId) {
+      return "shipping_quotes"
+    }
+    if (isArtsyShipping && selectedShippingQuoteId) {
+      return "ready_to_proceed"
+    }
+    if (!!savedFulfillmentType && !isArtsyShipping) {
+      return "ready_to_proceed"
+    }
+    return "fulfillment_details"
+  }, [isArtsyShipping, savedFulfillmentType, selectedShippingQuoteId])
+
+  return {
+    savedOrderData: savedOrderData,
+    initialValues,
+    step: targetStep,
+    helpers: {
+      fulfillmentDetails: useFulfillmentDetailsHelpers(),
+    },
+  }
+}
+
+const useFulfillmentDetailsHelpers = (): ShippingContextProps["helpers"]["fulfillmentDetails"] => {
+  // Because there is a single button for both fulfillment details and
+  // shipping quote steps (and duplicated in the sidebar)
+  // we need to hack some formik values UP from the fulfillment details
+  // form.
+  // handleSubmit: Used to submit the form
+  // isValid: Used to disable the button
+  // values: Used to get the form values for un-saving the address if the user
+  //   unchecks it after saving it in the fulfillment details step.
+  const [fulfillmentFormHelpers, setFulfillmentFormHelpers] = useState<
+    Pick<FormikProps<FulfillmentValues>, "handleSubmit" | "isValid" | "values">
+  >({
+    handleSubmit: () => {},
+    isValid: false,
+    values: {
+      attributes: {
+        saveAddress: false,
+      },
+    } as any,
+  })
+  return {
+    ...fulfillmentFormHelpers,
+    setFulfillmentFormHelpers,
+  }
+}
+
+const getInitialValues = (
```

Comment:
There's so much stuff here in this file that requires tests. We need to create a folder and structure things appropriately: 
```
src/Apps/Order/Routes/Shipping2/ShippingContext/ShippingContext.tsx
src/Apps/Order/Routes/Shipping2/ShippingContext/utils/foo.ts
src/Apps/Order/Routes/Shipping2/ShippingContext/utils/__tests__/foo.test.ts
src/Apps/Order/Routes/Shipping2/ShippingContext/hooks/useFoo.ts
src/Apps/Order/Routes/Shipping2/ShippingContext/hooks/__tests__/useFoo.test.ts
```

Reply:
yes i agree

Reply:
Totally. I think, too, now that everything is in this single file all all immediately referenceable (without needing to jump all around) we'll be able to do some further organization and from there and then, once its just right, break it apart (if need be).

Reply:
Ticketed in https://artsyproduct.atlassian.net/browse/EMI-1535.